src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -398,10 +419,24 @@ private void setState(State state) {
     this.state = state;
   }
 
+  public void setIsFirst(boolean isFirst) {
+    this.isFirst = isFirst;
+  }
+
+  public void setFlushPartial(boolean flushPartial) {
```

Comment:
@ewencp Sorry for late replying and thanks for the review.

I removed the flushPartial flag and the non-blackbox test case.

Regarding isFirst, it is kept with the following reason. 

In the shouldRotate method, there is a periodicRotation check on the time difference between 'now' and 'lastRotate' like so: now - lastRotate >= rotateIntervalMs. The 'lastRotate' is always zero because it is first used in shouldRorate() and will only be updated later on in updateRotationTimers(). 

Without the isFirst flag, the very first batch always contains only one record.